src/gui/qgsfixattributedialog.cpp

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+    qgsfixattributedialog.cpp
+    ---------------------
+    begin                : January 2020
+    copyright            : (C) 2020 by David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsfixattributedialog.h"
+
+#include "qgsattributeform.h"
+#include "qgsapplication.h"
+
+#include <QtWidgets/QPushButton>
+
+QgsFixAttributeDialog::QgsFixAttributeDialog( QgsVectorLayer *vl, QgsFeatureList &features, QWidget *parent )
+  : QDialog( parent )
+  , mFeatures( features )
+{
+  init( vl );
+}
+
+void QgsFixAttributeDialog::init( QgsVectorLayer *layer )
+{
+  QgsAttributeEditorContext context;
+  setWindowTitle( tr( "%1 - Fix Pastet Features" ).arg( layer->name() ) );
+  setLayout( new QGridLayout() );
+  layout()->setMargin( 0 );
+  context.setFormMode( QgsAttributeEditorContext::StandaloneDialog );
+
+  mUnfixedFeatures = mFeatures;
+  mCurrentFeature = mFeatures.begin();
+
+  QGridLayout *infoLayout = new QGridLayout();
+  QWidget *infoBox = new QWidget();
+  infoBox->setLayout( infoLayout );
+  layout()->addWidget( infoBox );
+
+  mDescription = new QLabel( descriptionText() );
+  mDescription->setVisible( mFeatures.count() > 1 );
+  infoLayout->addWidget( mDescription );
+  mProgressBar = new QProgressBar();
+  mProgressBar->setOrientation( Qt::Horizontal );
+  mProgressBar->setRange( 0, mFeatures.count() );
+  mProgressBar->setVisible( mFeatures.count() > 1 );
+  infoLayout->addWidget( mProgressBar );
+  QgsFeature feature;
+  mAttributeForm = new QgsAttributeForm( layer, *mCurrentFeature, context, this );
+  mAttributeForm->setMode( QgsAttributeEditorContext::SingleEditMode );
+  mAttributeForm->disconnectButtonBox();
+  layout()->addWidget( mAttributeForm );
+
+  QDialogButtonBox *buttonBox = mAttributeForm->findChild<QDialogButtonBox *>();
+  QPushButton *cancelAllBtn = new QPushButton( tr( "Cancel all" ) );
```

Comment:
Fix capitalisation for buttons (should be title case)

Reply:
Actually these strings are a bit confusing. I'd go with "discard all",  "discard all invalid features" and "paste all (including invalid)".

Reply:
@nirvn these buttons are very long - got any ideas on alternative ui?